components/fxa-client/src/migrator.rs

```diff
@@ -25,17 +25,15 @@ impl FirefoxAccount {
             return Err(ErrorKind::IllegalState("Session Token is already set.").into());
         }
         // Trade our session token for a refresh token.
-        self.state.session_token = Some(session_token.to_string());
```

Comment:
This was stashing *Fennec*'s sesstionToken in the state, whereas IIUC it should be stashing the duplicate one with we made for our own purposes.

Reply:
Nice find